multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakSettings.java

```diff
@@ -162,17 +162,17 @@ public KeycloakSettings(
     if (userInfoEndpoint != null) {
       settings.put(USERINFO_ENDPOINT_SETTING, userInfoEndpoint);
       if (serverURL != null) {
-        String internalInfoEndpoint = userInfoEndpoint.replace(serverURL, serverInternalURL);
-        internalSettings.put(USERINFO_ENDPOINT_SETTING, internalInfoEndpoint);
+        userInfoEndpoint = userInfoEndpoint.replace(serverURL, serverInternalURL);
       }
+      internalSettings.put(USERINFO_ENDPOINT_SETTING, userInfoEndpoint);
     }
     String jwksUriEndpoint = (String) openIdConfiguration.get("jwks_uri");
     if (jwksUriEndpoint != null) {
       settings.put(JWKS_ENDPOINT_SETTING, jwksUriEndpoint);
       if (serverURL != null) {
-        String internalJwksUriEndpoint = jwksUriEndpoint.replace(serverURL, serverInternalURL);
-        internalSettings.put(JWKS_ENDPOINT_SETTING, internalJwksUriEndpoint);
+        jwksUriEndpoint = jwksUriEndpoint.replace(serverURL, serverInternalURL);
       }
+      internalSettings.put(JWKS_ENDPOINT_SETTING, jwksUriEndpoint);
```

Comment:
can we extract it to the `private String getjwksUriEndpoint()`;
Also, tests will be much appreciated ;-)

Reply:
@ibuziuk 

@AndrienkoAleksandr is on PTO today. He will take a look tomorrow.

Reply:
ok, thanks - I'm ok if we merge it in the current state as an exception, otherwise, we can have a fix backported to `7.23.x` and wait for `7.23.1` for Hosted Che update (currently this is a blocker for the latest version rollout)